src/main/java/com/ot/restaurant/ioc/DependencyContainer.java

```diff
@@ -0,0 +1,54 @@
+package com.ot.restaurant.ioc;
+
+import java.io.File;
+
+public class DependencyContainer<T> {
+  private final IoCContainer ioc = IoCContainer.getInstance();
+
+  private final Object entityType = new Object();
+
+  public DependencyContainer() {}
+
+  public void registerRepositories() throws Exception {
```

Comment:
Do this manaually please, what is we create UserRepositoryV2Impl class? this could make a mistake, we should be able to decide which instance to use under the same ioc container name